### PR TITLE
feat(analysis): Adds timeout property to NewRelic metrics provider. Resolves: #3741

### PR DESCRIPTION
**Pull request by orlando-valdez-ck**
Resolves: #3741 

Adds a timeout property to the New Relic metrics provider spec to be sent to the NerdGraph API for queries that take longer than the default 5 seconds. I bumped up the newrelic-go-client library while working on this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: metricproviders/newrelic/newrelic.go [hunk 1]
```diff
@@ -41,13 +66,30 @@ type NewRelicClient struct {
 }
 
 // Query executes a NRQL query against the given New Relic account
-func (n *NewRelicClient) Query(query string) ([]nrdb.NRDBResult, error) {
-	results, err := n.Nrdb.Query(n.AccountID, nrdb.NRQL(query))
```
Comment (orlando-valdez-ck) [inline metricproviders/newrelic/newrelic.go:45]: There's an equivalent [function](https://github.com/newrelic/newrelic-client-go/blob/36d7acf42b8bd408e77dd1898ab148ee43922a15/pkg/nrdb/nrdb_api.go#L107) that supports passing timeout in the `nrdb` package but it [pulls way more data](https://github.com/newrelic/newrelic-client-go/blob/36d7acf42b8bd408e77dd1898ab148ee43922a15/pkg/nrdb/nrdb_api.go#L213) than the current function which by itself can impact latency.
That's why I switched to the NerdGraph package to send the timeout in the query without incurring in additional data over the wire. I actually slimmed down the query compared to the [used](https://github.com/newrelic/newrelic-client-go/blob/36d7acf42b8bd408e77dd1898ab148ee43922a15/pkg/nrdb/nrdb_api.go#L40) in the `Query` function to just what we need for this metric provider so the GQL API has to do less work to respond (and even less data transmitted back)

**Comment by zachaller** [comment]
Can I have you update with master and fix conflicts, that should re-run the tests now that I have them fixed.

**Comment by orlando-valdez-ck** [comment]
Noticed something during my last round of local testing. Changed to draft while I figure this out.

**Comment by orlando-valdez-ck** [comment]
Ready for review now

**Review by zachaller — APPROVED** [review]

**Pull request merged**